p2p/libp2p/netMessenger.go

```diff
@@ -130,39 +139,78 @@ func NewNetworkMessenger(args ArgsNetworkMessenger) (*networkMessenger, error) {
 		return nil, err
 	}
 
-	port, err := getPort(args.P2pConfig.Node.Port, checkFreePort)
+	setupExternalP2PLoggers()
+
+	h, ctx, cancelFunc, port, err := createHostWithRetry(args, p2pPrivKey, reusePort)
 	if err != nil {
 		return nil, err
 	}
 
+	p2pNode, err := createMessenger(args, h, ctx, cancelFunc, port, withMessageSigning)
+	if err != nil {
+		log.LogIfError(h.Close())
+		return nil, err
+	}
+
+	return p2pNode, nil
+}
+
+func createHost(args ArgsNetworkMessenger, p2pPrivKey *libp2pCrypto.Secp256k1PrivateKey, portReuse bool) (host.Host, context.Context, func(), int, error) {
```

Comment:
too many returned stuff. maybe use a DTO for returning and you'd use something like `(createHostResults, error)`

Reply:
refactored